### PR TITLE
Fixes #1115: Make GenericScraper.java functional

### LINKED ISSUE #1115: GenericScraper is not working
The dependency of GenericScraper.java , i.e. Boilerpipe does not exist in the project, so this doesn't work.

Output of GenericScraper :-

screenshot from 2017-04-29 23-45-42

### PR DESCRIPTION
**Pull request by vibhcool**
This PR :-
 - Fixes #1115 adds boilerpipe library to make GenericScraper.java functional
 - add more boilerpipe extractors in GenericScraper.java
 - make DefaultExtractor as default type in GenericScraper.java

Example links:-

http://localhost:9000/api/genericscraper.json?url=https://github.com
http://localhost:9000/api/genericscraper.json?type=default&url=https://github.com
http://localhost:9000/api/genericscraper.json?type=all&url=https://github.com
http://localhost:9000/api/genericscraper.json?type=article&url=https://github.com
http://localhost:9000/api/genericscraper.json?type=main&url=https://github.com
http://localhost:9000/api/genericscraper.json?type=canola&url=https://github.com

Example Output
screenshot from 2017-04-30 04-47-30

### CODE REVIEW

**Comment by vibhcool** [comment]
tested and open for review :)

File: build.gradle [hunk 1]
```diff
@@ -125,6 +125,7 @@ dependencies {
   compile group: 'org.twitter4j', name: 'twitter4j-core', version: '4.0.2'
   compile group: 'org.twitter4j', name: 'twitter4j-stream', version: '4.0.2'
   compile group: 'xerces', name: 'xerces', version: '2.4.0'
+  compile fileTree(dir: 'libs', include: ['*.jar'])
```
Comment (sudheesh001) [inline build.gradle]: The intention over time was to remove the `libs/` folder. I notice that you've added boilerpipe to this, There are some problems with that library. It's not being maintained anymore and a better thing to do right now is to take the maintainer status of that project onto the loklak org and publish a gradle entry and compile via the gradle name.

File: src/org/loklak/api/search/GenericScraper.java [hunk 2]
```diff
@@ -97,22 +99,47 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
...
+            case "canola":
+                genericScraperData = bpipeExtract(url, genericScraperData, CommonExtractors.CANOLA_EXTRACTOR);
+                printJSON(response, genericScraperData);
+                break;            
+            // Returns error message for wrong type
+            default:
+                type = "error";
+                genericScraperData.put("error", "Please mention type of scraper: <type> ('article', 'default', 'main', 'all', 'canola')");
+            	printJSON(response, genericScraperData);
+                break;
```
Comment (sudheesh001) [inline src/org/loklak/api/search/GenericScraper.java]: It'd be a nice thing to have these extractors within loklak and then use it. I think to perform this patch efficiently we need to either merge the boilerpipe code with our existing structure and then work on it, thereby leaving us the `CommonExtractors` to work with and enhance.

**Comment by sudheesh001** [comment]
Also commit message needs corrections. Link to issue #1116

File: build.gradle (hunk 1 above)
Comment (yukiisbored) [inline build.gradle]: Please don't add the ungodly libs folder.

Comment (vibhcool) [inline build.gradle]: @sudheesh001 @yukiisbored ok, got it, I will keep these points in mind.

I will try to go through the boilerpipe library and try to implement it's logic to be able to merge some parts of it with genericscraper . 

For time being I shall close this PR

**Pull request closed without merging**